components/script/script_thread.rs

```diff
@@ -3043,20 +3048,6 @@ impl ScriptThread {
         }
     }
 
-    fn ask_constellation_for_browsing_context_info(
```

Comment:
Note: why did I remove the method and move its body to the point of use? Two reasons on my side:

1. It's only used in one place.
2. I think it's an anti-pattern so I'd rather not have it available as a method for use in other places. 

Please let me know what you think.

Reply:
I aslo tried to remove this sync comm with the constellation, by switching to using a BC id as the "parent info" all across, but it broke a few things, I think perhaps because I would rely on the "active" doc of the window proxy to grab the pipeline id when necessary, and the concept of "active doc" is unlikely to competely overlap with the use of the pipeline id now. It's at https://github.com/servo/servo/commit/ada2f9c7e41dc49455df03e80ff1ed0235e64b0f